### DISCUSSION TITLE
How to query based on secondary partition key with DynamoDbTemplate?

### DISCUSSION THREAD

**Discussion by johnsall** [awspring/spring-cloud-aws#683] [Q&A]
I have a Posts Kotlin class like so:

```@DynamoDbBean
data class Posts(
    @get:DynamoDbPartitionKey
    @get:DynamoDbAttribute("post_id")
    var id: String = UUID.randomUUID().toString(),
    
    @get:DynamoDbAttribute("user_id")
    @get:DynamoDbSecondaryPartitionKey(indexNames = ["post_user_id_created_at_idx"])
    var userId: String = "",

    @get:DynamoDbAttribute("title")
    @get:DynamoDbSecondaryPartitionKey(indexNames = ["post_title_created_at_idx"])
    var title: String = "",

    @get:DynamoDbAttribute("content")
    var content: String = "",

    @get:DynamoDbAttribute("created_at")
    @get:DynamoDbSortKey
    @get:DynamoDbSecondarySortKey(indexNames = ["post_user_id_created_at_idx", "post_title_created_at_idx"])
    var createdAt: LocalDateTime = LocalDateTime.now()
)
```

How can I use the DynamoDbTemplate to query data compared against the user_id or the title? I'm using the 3.0.0-RC1 version of the library as I need compatibility with SpringBoot 3.

Thank you!

**Comment by MatejNedic** [comment]
Hey @johnsall , currently with 3.0.0-RC1 it is not possible. However https://github.com/awspring/spring-cloud-aws/pull/680 has resolved this and with a new release it will be possible! Hopefully soon.

**Reply by johnsall** [reply]
Great news @MatejNedic!! Are there any news on having the Enhanced client be Async?

**Reply by MatejNedic** [reply]
@johnsall currently not, but open an issue and we can implement it :)
